testing/example_scripts/issue_519.py

```diff
@@ -22,13 +22,13 @@ def checked_order():
     assert order == [
         ("issue_519.py", "fix1", "arg1v1"),
         ("test_one[arg1v1-arg2v1]", "fix2", "arg2v1"),
-        ("test_two[arg1v1-arg2v1]", "fix2", "arg2v1"),
```

Comment:
Hmm, this is odd, because we have increased the number of setups/teardowns for fixture 2 with this change.

On `main`, we have 2 s/t for `fix1`, and 4 s/t for `fix2`.

Here we have the same 2 s/t for `fix1`, but now we have 8 s/t for `fix2`. Seems like a regression for this case. 🤔

Reply:
> now we have 8 s/t for fix2

Yes, but we also had 8 s/t for `fix2` even before this PR. `fix2` is always (and correctly) recreated, because it has function scope. It's just not obvious from the assert statement. Tested it by adding print output pre and post the yield in `foo{1,2}`. But please double check...

Reply:
I think the same holds for fixture `arg2`, it also has function scope and therefore shall always be recreated.

`fix1` and `arg1` have module scope, and both are successfully reused even after the PR, are they?

Reply:
Ahh right my bad, missed the fact that `fix2` is function scoped, so its order there doesn't matter as it will be recreated each time. 👍